src/vs/workbench/browser/layout.ts

```diff
@@ -366,6 +392,49 @@ export abstract class Layout extends Disposable implements IWorkbenchLayoutServi
 		this.layout();
 	}
 
+	private setWindowBorder(skipLayout: boolean = false) {
+		if (isWeb || getTitleBarStyle(this.configurationService, this.environmentService) !== 'custom') {
+			return;
+		}
+
+		const theme = this.themeService.getTheme();
+
+		const activeBorder = theme.getColor(WINDOW_ACTIVE_BORDER);
+		const inactiveBorder = theme.getColor(WINDOW_INACTIVE_BORDER);
+
+		let windowBorder = false;
+		if (activeBorder || inactiveBorder) {
+			windowBorder = true;
+
+			let borderColor = this.state.hasFocus ? activeBorder : inactiveBorder;
+			if (!borderColor) {
+				borderColor = theme.getColor(this.state.hasFocus ? TITLE_BAR_ACTIVE_BACKGROUND : TITLE_BAR_INACTIVE_BACKGROUND);
+			}
+			this.container.style.setProperty('--window-border-color', borderColor ? borderColor.toString() : 'transparent');
+		}
+
+		if (windowBorder === this.state.windowBorder) {
+			return;
+		}
+
+		this.state.windowBorder = windowBorder;
+
+		if (windowBorder) {
+			addClass(this.container, Classes.WINDOW_BORDER);
+		}
+		else {
```

Comment:
Our coding style is to put the `else` statement on the same line. Even better, I suggest to use `toggleClass` helper to remove the if/else.